homeassistant/components/fritzbox/binary_sensor.py

```diff
@@ -68,18 +68,25 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the FRITZ!SmartHome binary sensor from ConfigEntry."""
-    coordinator: FritzboxDataUpdateCoordinator = hass.data[FRITZBOX_DOMAIN][
-        entry.entry_id
-    ][CONF_COORDINATOR]
-
-    async_add_entities(
-        [
-            FritzboxBinarySensor(coordinator, ain, description)
-            for ain, device in coordinator.data.devices.items()
-            for description in BINARY_SENSOR_TYPES
-            if description.suitable(device)
-        ]
-    )
+    coordinator: FritzboxDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        CONF_COORDINATOR
+    ]
+
+    def _add_entities() -> None:
+        """Add devices."""
+        async_add_entities(
+            [
+                FritzboxBinarySensor(coordinator, ain, description)
+                for ain, device in coordinator.data.devices.items()
+                if ain in coordinator.new_devices
```

Comment:
You could keep a local list in this platform with added devices. Now its somewhat racy for coordinator to know when to clear that list.

Reply:
the coordinator clears this list at every data update run, the platforms only check this list, after the update run has completed, so this list should always be up-to-date and platforms should only "see" really new devices.
hopefully i do not miss something 🤔

Reply:
The signal listeners may not execute right away. So the signaller will run async from this and might end up updating again (somewhat unlikely but possible)